packages/slate/src/utils/get-normalize-setting.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Obtains the normalization setting to use for the change based on the
+ * change object flags and the change operation options.
+ *
+ * @param {Object} flags - flags object for the current change object
+ * @param {Object} options - options for the change operation
+ * @return {bool} - the normalization configuration to use. Defaults to true
+ *   if flags or options does not have a normalization configuration
+ */
+
+function getNormalizeSetting(flags, options) {
+  let normalize = true
```

Comment:
I think this value should then be set as the default on `this.flags` in the `Change` objects? So that there isn't a hardcoded one here?

Reply:
Hm, this one might require a little bit more careful examination (and might cause some complications with the lovely `getFlag` idea).

The default here comes from the `const { normalize = true } = options` which would set normalize to `true` if it was absent from `options`. 

The existing implementation of this function is set up carefully to have the following logic

1. Assume the normalize flag is true.
2. Check if normalize flag is set in operation flags. If so return that value. This is to allow flags to have the highest precedence for this normalization configuration.
3. Finally check if options has set normalize. If so, use that value only if normalize is not specified in flags.

This is important because we don't want folks doing something odd like setting `{ normalize: true }` in a change call and breaking the contract of `withoutNormalization`

Reply:
@ianstormtaylor  alright! I think I got it sorted out. Going to write some comments on the new getFlag method and start a discussion for one of the calls